OrbitService/ProcessServiceImpl.cpp

```diff
@@ -41,8 +46,12 @@ Status ProcessServiceImpl::GetModuleList(ServerContext*,
   int32_t pid = request->process_id();
   LOG("Sending modules for process %d", pid);
 
-  std::vector<ModuleInfo> module_infos = LinuxUtils::ListModules(pid);
-  for (const auto& module_info : module_infos) {
+  const auto module_infos = LinuxUtils::ListModules(pid);
+  if (!module_infos) {
+    return Status(StatusCode::NOT_FOUND, module_infos.error().message());
```

Comment:
with current way of handling errors this will return quite a cryptic message to a user... 
What I would like to see here is something along the lines "Unable to read /proc/<pid>/maps: Permission denied" - this gives an idea to user and us of what happened.

Reply:
Note also that our users are Software Engineer - they will likely appreciate details in the error messages.

Reply:
I made the error message more expressive now by using std::string as error type in ListModules. I filed a bug regarding unified error handling b/160554376